.github/workflows/clean_dedicated_host.yml

```diff
@@ -0,0 +1,38 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+
+name: Clean Dedicated Host Daily Resources Cleaner
+
+on:
+  schedule:
+    - cron: "0 0 * * *" # Run Every Day At Midnight
+  workflow_dispatch:
+
+jobs:
+  clean-dedicated-hosts:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v3
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.TERRAFORM_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.TERRAFORM_AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+
+      - name: Clean old dedicated host
+        run: go run ./integration/clean/clean_dedicated_host/clean_dedicated_host.go --tags=clean
+
+      # @TODO delete this when we remove the internal cw agent integration pipeline
+      # In the mean time just set this secret to your account creds
+      - name: Configure AWS Credentials
```

Comment:
IDK how granular we can get with configurations, but is there a way that we could specify that this specific step should only run in **this** repo, and not on people's forks? I know we can configure triggers to only work on specific repos, but not sure about individual steps.

Reply:
If not, is there a way that we could add a conditional here, like "if INTERNAL_AWS_ACCESS_KEY_ID is nonempty, run this step"? 

Basically, I just don't want to set up another credential on my fork if I don't have to.

Reply:
The workflow is not valid. .github/workflows/clean_dedicated_host.yml (Line: 30, Col: 13): Unrecognized named-value: 'secrets'. Located at position 1 within expression: secrets.INTERNAL_AWS_ACCESS_KEY_ID .github/workflows/clean_dedicated_host.yml (Line: 38, Col: 13): Unrecognized named-value: 'secrets'. Located at position 1 within expression: secrets.INTERNAL_AWS_ACCESS_KEY_ID
 I'm just going to move this out to another workflow for now